Library/Homebrew/bottles.rb

```diff
@@ -39,11 +38,18 @@ def bottle_tag
   end
 end
 
-def bottle_filename_formula_name filename
-  path = Pathname.new filename
-  version = BottleVersion.parse(path).to_s
-  basename = path.basename.to_s
-  basename.rpartition("-#{version}").first
+def bottle_formula_name bottle_file
+  receipt_file_path = Utils.popen_read("tar", "-tzf", bottle_file, "*/*/INSTALL_RECEIPT.json").chomp
+  receipt_file = Utils.popen_read("tar", "-xOzf", bottle_file, receipt_file_path)
+  name = receipt_file_path.split("/").first
+  tap = Tab.from_tabfile_content(receipt_file, "#{bottle_file}/#{receipt_file_path}").tap
+
+  if tap.nil? || tap == "Homebrew/homebrew"
+    full_name = name
+  else
+    full_name = "#{tap.sub("homebrew-", "")}/#{name}"
+  end
+  [name, full_name]
```

Comment:
I don't think this should return two arguments without changing the function name to indicate that.